CRM/Contribute/BAO/Contribution.php

```diff
@@ -2715,20 +2715,21 @@ public function _assignMessageVariablesToTemplate(&$values, $input, $returnMessa
       $softDAO = new CRM_Contribute_DAO_ContributionSoft();
       $softDAO->contribution_id = $this->id;
       if ($softDAO->find(TRUE)) {
-        $pcpParams['pcpBlock'] = TRUE;
         $pcpParams['pcp_display_in_roll'] = $softDAO->pcp_display_in_roll;
         $pcpParams['pcp_roll_nickname'] = $softDAO->pcp_roll_nickname;
         $pcpParams['pcp_personal_note'] = $softDAO->pcp_personal_note;
 
         //assign the pcp page title for email subject
-        $pcpDAO = new CRM_PCP_DAO_PCP();
-        $pcpDAO->id = $softDAO->pcp_id;
-        if ($pcpDAO->find(TRUE)) {
-          $pcpParams['title'] = $pcpDAO->title;
-
-          // do not display PCP block in receipt if not enabled for the PCP poge
-          if (empty($pcpDAO->is_honor_roll)) {
-            $pcpParams['pcpBlock'] = FALSE;
+        if ($softDAO->pcp_id) {
+          $pcpDAO = new CRM_PCP_DAO_PCP();
+          $pcpDAO->id = $softDAO->pcp_id;
+          if ($pcpDAO->find(TRUE)) {
+            $pcpParams['title'] = $pcpDAO->title;
+
+            // Only display PCP block in receipt if enabled for the PCP page
+            if (!empty($pcpDAO->is_honor_roll)) {
+              $pcpParams['pcpBlock'] = TRUE;
+            }
```

Comment:
@demeritcowboy I am just wondering is pcpBlock assigned as FALSE by default at all anywhere? just thinking it would be good to have it always assigned to the template

Reply:
up a few lines it is set to null